### PR TITLE
Add `occ user:welcome` command to send user welcome email from CLI

### LINKED ISSUE #39610: Add user:welcome command
### How to use GitHub

* Please use the 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to show that you are interested into the same feature.
* Please don't comment if you have no relevant information to add. It's just extra noise for everyone subscribed to this issue.
* Subscribe to receive notifications on status change and new comments. 

**Is your feature request related to a problem? Please describe.**
When I create users from occ I would like to send a welcome email with the reset password link.

**Describe the solution you'd like**
```
php occ user:welcome username --reset-password
```

**Describe alternatives you've considered**
N/A

**Additional context**
N/A

### PR DESCRIPTION
**Pull request by FedericoHeichou**
* Resolves: #39610 

## Summary
Implements `user:welcome` command.  
I tested in 26 and master

## TODO

- [ ] ...

## Checklist

- Code is [properly formatted](https://docs.nextcloud.com/server/latest/developer_manual/digging_deeper/continuous_integration.html#linting)
- [Sign-off message](https://github.com/src-d/guide/blob/master/developer-community/fix-DCO.md) is added to all commits
- [ ] Tests ([unit](https://docs.nextcloud.com/server/latest/developer_manual/app_development/tutorial.html#unit-tests), [integration](https://docs.nextcloud.com/server/latest/developer_manual/app_development/tutorial.html#integration-tests), api and/or acceptance) are included
- [ ] Screenshots before/after for front-end changes
- [ ] Documentation ([manuals](https://github.com/nextcloud/documentation/) or wiki) has been updated or is not required
- [x] [Backports requested](https://github.com/nextcloud/backportbot/#usage) where applicable (ex: critical bugfixes)

### CODE REVIEW

**Comment by tanganellilore** [comment]
Hi team,
I'm also interested to this features... @solracsf @icewind1991 can you help us to address this and include in 28?

Thanks

File: core/Command/User/Welcome.php [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+		IUserManager $userManager,
+		NewUserMailHelper $newUserMailHelper
+	) {
+		parent::__construct();
+
+		$this->userManager = $userManager;
+		$this->newUserMailHelper = $newUserMailHelper;
+	}
+
+	protected function configure() {
```
[the remark this answers is not in the document]
Comment (FedericoHeichou) [reply core/Command/User/Welcome.php:59]: Done

**Review by Altahrim — APPROVED** [review]

**Comment by Altahrim** [comment]
@FedericoHeichou Would you mind rebasing the PR and rewrite the commit with [conventional commits](https://www.conventionalcommits.org/)?

**Review by sorbaugh — APPROVED** [review]

**Comment by sorbaugh** [comment]
Hello @FedericoHeichou, going through the PRs this one looks quite nice. I'm reopening and would like to help you get this in.

**Comment by sorbaugh** [comment]
We try our best to keep our PR list "up to date" by cleaning up PRs that haven't seen activity in a while (many PRs do get stale/abandoned unfortunately). We do go over PRs that get "caught" in this and push them forward again. Hope this explains a bit!

File: core/Command/User/Welcome.php [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
```
Comment (skjnldsv) [inline core/Command/User/Welcome.php:23]:
```suggestion
/**
 * SPDX-FileCopyrightText: 2023 FedericoHeichou <federicoheichou@gmail.com>
 * SPDX-License-Identifier: AGPL-3.0-or-later
 */
```

**Review by skjnldsv — APPROVED** [review]

**Comment by skjnldsv** [comment]
Looking good!! Let's merge!! :rocket:!! Thank you for the patience and the great feature @FedericoHeichou :bow: :tada:

**Comment by ShGKme** [comment]
@sorbaugh Can we backport it to 30?

**Pull request merged**